### ISSUE TITLE
Feature request: Add radiobuttons() funcitonality to the package?

### ISSUE THREAD

**Issue by TarJae** [priism-center/shinyQuiz#3]
Hi. 
Congratulations on the great package. 
I'm not very skilled in dealing with GitHub and pull requests, etc... I need the same functionality as shinyquiz but only with radio buttons. The background is to design a vocabulary test app. Here, only radio buttons can be used. In principle, the same functionality as `selectInput()` but preferring radio buttons instead. 

I don't know whether to do a pull request or create a new app with your code, or simply make a feature request like this. Thank you for your attention and processing.

This manipulation of your code fulfill my needs: 

## Adapt `create_question_input_` function in `creators.R`
```
create_question_input_ <- function(dot_list, choices, type, input, label, selected, ns){
  
  # Extract answer texts
  texts <- purrr::map(choices, \(x) x@text)
  
  # Check which choice is correct
  is_correct <- purrr::map_lgl(choices, \(x) x@correct)
  if (sum(is_correct) < 1) stop('Choices must contain at least one correct answer', call. = FALSE)
  text_correct <- texts[is_correct]
  
  # Select type and input type
  if (type == 'auto'){
    type <- ifelse(sum(is_correct) == 1, 'single', 'multiple')
  }
  if (input == 'auto'){
    input <- ifelse(type == 'single', 'radio', 'checkbox')  # Default to radio for single choice
  }
  
  # Determine the input method
  if (input == 'radio') {
    input_html <- shiny::radioButtons(
      inputId = ns('answers'), 
      label = label,
      choices = setNames(texts, texts),  # Associate each choice with its text for clarity
      selected = selected
    )
  } else if (input == 'checkbox') {
    input_html <- shiny::checkboxGroupInput(
      inputId = ns('answers'), 
      label = label,
      choices = setNames(texts, texts),  # Similarly, using setNames for clarity
      selected = selected
    )
  } 
  else {
    stop('Unsupported input type', call. = FALSE)
  }
  
  
  return(list(input_html = input_html, text_correct = text_correct))
}
```
and  in `Create_question` function 2 things: 
```
create_question <- function(prompt, ..., type = c('auto', 'single', 'multiple'), input = c('auto', 'select', 'checkbox', 'radio'), shuffle = FALSE, ns = shiny::NS('quiz')){
```
and : 
```
input <- match.arg(input, c('auto', 'select', 'checkbox', 'radio'))
```
and to test: 
add to `example-app.R`
```
....
quiz <- create_quiz(
  create_question(
    'Select C.',
    add_choice('A'),
    add_choice('B'),
    add_choice('C', correct = TRUE)
  ),
  create_question(
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed id ornare augue, fringilla molestie metus. Donec eget tortor tincidunt, sagittis dui volutpat, finibus est. Select nulla.',
    add_choice('Nulla vel'),
    add_choice('auctor nulla'),
    add_choice('nulla', correct = TRUE)
  ),
  .....
```

**Comment by joemarlo** [comment]
Great idea. I can't believe this isn't in the package already. Those look like most or all of the right places to implement the change; I'll take a deeper look later this week when I have time. In the meantime, you can also use `create_question_raw()` to implement radio buttons. Here's a reprex:

```
library(shinyquiz)

quiz <- create_quiz(
  create_question_raw(
    prompt = htmltools::div(
      htmltools::p("What is the capital of Illinois?"),
      shiny::radioButtons(
        inputId = shiny::NS('quiz')('answers'),
        label = 'Select one',
        choices = c('Chicago', 'Paw Paw', 'Springfield')
      )
    ),
    grader = \(user_input) user_input == 'Springfield',
    correct_answer_pretty = 'Springfield'
  ),
  create_question(
    'Which elements are gases at room temperature? Select all that apply.',
    add_choice('Hydrogen', correct = TRUE),
    add_choice('Mercury'),
    add_choice('Nitrogen', correct = TRUE),
    label = 'Select Hydrogen and Nitrogen'
  ),
  create_question(
    'At what temperature does water boil at sea level?',
    add_slider(min = 90, max = 150, default_position = 120, correct = 100),
    label = 'Select 100'
  )
)
preview_app(quiz)
```

**Comment by joemarlo** [comment]
Implemented in [commit cb1c0dfed2f06ab2cd6aeb5d0fc6c33096cf8693](https://github.com/priism-center/shinyquiz/commit/cb1c0dfed2f06ab2cd6aeb5d0fc6c33096cf8693)
